src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -92,6 +92,7 @@ template Instruction
             vSenderAccount <- view <$> fetchInterfaceByKey @Account.R senderAccount
             assertMsg "actors must include instructors" $ S.isSubsetOf vSenderAccount.controllers.instructors actors
             assertMsg "actors must include lockers" $ S.isSubsetOf (Lockable.getLockers lockable) actors
+            assertMsg "the custodian of the pledged asset does not match the target custodian" $ step.custodian == senderAccount.custodian
```

Comment:
maybe formulate it as "the custodian of the pledged asset must match the target custodian"